rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -89,6 +91,72 @@ def _store_ca(cls, service_type, service_node):
         return ca_path
 
 
+class ConcurrentStatusChecker(object):
+    STOP_THE_WORKER = 'STOP THE WORKER'
+
+    def __init__(self, number_of_threads=3):
+        self._number_of_threads = number_of_threads
+        self._in_queue = queue.Queue()
+        self._in_queue_len = 0
+        self._out_queue = queue.Queue()
+        self._result_cache = {}
+        self._threads = [
+            threading.Thread(target=self._worker)
+            for _ in range(self._number_of_threads)
+        ]
+
+    def _worker(self):
+        while True:
+            service_type, service_node, service_node_name, status_func = \
+                self._in_queue.get()
+            if service_type == ConcurrentStatusChecker.STOP_THE_WORKER:
+                break
+            creds = Credentials.get(service_type, service_node.private_ip)
+            try:
+                prometheus_response = status_func(
+                    'https://{ip}:53333/monitoring/'.format(
+                        ip=service_node.private_ip),
+                    auth=(creds.username, creds.password),
+                    ca_path=creds.ca_path,
+                )
+            except Exception:
+                self._out_queue.put((service_node_name, [],))
+            else:
+                self._out_queue.put((service_node_name,
+                                     prometheus_response,))
+
+    def append(self, service_type, service_node, service_node_name,
+               status_func):
+        self._in_queue.put((service_type, service_node, service_node_name,
+                            status_func,))
+        self._in_queue_len += 1
+        self._result_cache[service_node_name] = []
+        for thread in [t for t in self._threads if not t.is_alive()]:
```

Comment:
doing this in `append` is a bit weird. We don't need the threads in append, really, we need them in get_responses, for getting the responses.
How about, in `__init__`, `self._threads = None` and then in `get_responses`,  `if self._threads is None:  self._threads = [...` and also start them right there